system/modules/core/classes/Backend.php

```diff
@@ -801,17 +795,20 @@ public static function addFileMetaInformationToRequest($strUuid, $strPtable, $in
 				break;
 		}
 
-		if ($objPage === null || $objPage->numRows < 1)
+		if ($objPage === null || ($objPage instanceof \Database\Result && $objPage->numRows < 1))
 		{
 			return;
 		}
 
-		$objModel = new \PageModel();
-		$objModel->setRow($objPage->row());
-		$objModel->loadDetails();
+		if (!$objPage instanceof \PageModel)
+		{
+			$objPage = \PageModel::findByPk($objPage->id);
```

Comment:
theoretically, but very unlikely, the ID could not exist. Your changes are fine to me 👍